### ISSUE TITLE
Discussion: Support toMaybe for Maybe Monad

### ISSUE THREAD

**Issue by martinlechner1** [ramda/ramda-fantasy#132]
I miss good way to wrap nullable values in the Maybe Monad. 
Option Types nowadays usually provide a static constructor ofNullable, which returns Just(x) or Nothing() depending on the value given.
I don't want to use Maybe(x) directly, since I want to show more explicitly that my wrapped value is nullable.
I could implement ofNullable if there is an agreement to support this.

**Comment by scott-christopher** [comment]
I'd be in favour of moving the current logic from the constructor to a static method.

**Comment by davidchambers** [comment]
We could name the function `toMaybe` to match [`S.toMaybe`][1].


[1]: https://sanctuary.js.org/#toMaybe

**Comment by TheLudd** [comment]
I would prefer a function `nothingIf`

```
nothingIf:: (a -> Boolean) -> a -> Maybe[a]
```
Now you can do `nothingIf(isNil)` and you have your `Maybe` factory.
There could also be a `justIf` function that works the other way around and creates a `Nothing` if the predicate function evaluates to `false`.

**Comment by davidchambers** [comment]
I don't see a need for `nothingIf`, @TheLudd. We can use [`filter`][1] or [`filterM`][2] instead:

```javascript
const S = require('sanctuary');
const Z = require('sanctuary-type-classes');

Z.filter(S.even, Just(41));  // => Nothing
Z.filter(S.even, Just(42));  // => Just(42)
Z.filterM(S.even, Just(41));  // => Nothing
Z.filterM(S.even, Just(42));  // => Just(42)
```


[1]: https://github.com/sanctuary-js/sanctuary-type-classes#filter
[2]: https://github.com/sanctuary-js/sanctuary-type-classes#filterM

**Comment by TheLudd** [comment]
> I miss good way to wrap nullable values in the Maybe Monad.

I understand this as wrapping a "primitive" value, i.e. not a `Maybe`, in a maybe. I don't see that in your example.

**Comment by davidchambers** [comment]
I see now that I wasn't clear, @TheLudd. What I'm suggesting is unconditionally applying `Just` to a value of type `a` to produce a value of type `Maybe a`, then filtering to replace an unwanted value with `Nothing`.

```javascript
Z.filter(S.even, Just(x));  // => Just(x) or Nothing, depending on x
```

One could define `nothingIf` in terms of these simpler functions:

```javascript
//    nothingIf :: (a -> Boolean) -> a -> Maybe a
const nothingIf = pred => x => Z.filter(R.complement(pred), Just(x));
```

**Comment by scott-christopher** [comment]
Implemented via #133 / #159.
